iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -85,11 +88,25 @@ private static void checkLinkedServices(ContextualResource resource) {
     ContextualProperty authenticationType = resource.object("typeProperties").property("authenticationType");
 
     if (type.is(isValue(SENSITIVE_LINKED_SERVICES_TYPE::contains))
-      && authenticationType.is(isValue(str -> !"ClientCertificate".equals(str)))) {
+      && authenticationType.is(isValue(str -> !CLIENT_CERTIFICIATE_VALUE.equals(str)))) {
       authenticationType.report(WRONG_AUTHENTICATION_METHOD_MESSAGE, type.toSecondary("Service type"));
     }
   }
 
+  private static void checkJobCollections(ContextualResource resource) {
+    resource.object("action")
+      .object("request")
+      .object("authentication")
+      .property("type")
+      .reportIf(isValue(str -> !CLIENT_CERTIFICIATE_VALUE.equals(str)), WRONG_AUTHENTICATION_METHOD_MESSAGE);
+    resource.object("action")
+      .object("errorAction")
+      .object("request")
+      .object("authentication")
+      .property("type")
+      .reportIf(isValue(str -> !CLIENT_CERTIFICIATE_VALUE.equals(str)), WRONG_AUTHENTICATION_METHOD_MESSAGE);
```

Comment:
As the last 4 lines of each report are identical, I would extract them so we don't have duplicates.